content/en/apps/guides/data/analytics/building-dbt-models.md

```diff
@@ -0,0 +1,376 @@
+  date_trunc('month', pregnancy.reported) as report_month
+FROM
+  pregnancy
+INNER JOIN contact chw ON chw.uuid = data_record.contact_uuid
+LEFT JOIN contact area ON area.uuid = data_record.parent_contact_uuid
+GROUP BY
+  chw.uuid,
+  chw.name,
+  area.uuid,
+  area.name,
+  report_month;
+```
+
+### Aggregations
+It may be useful to save some common queries as views, so that they can be reused in dashboards or downstream aggregations.
+With the pregnancy form model above, if there was also a postnatal care form they could be joined to view outcomes together with registrations.
+
+```sql
+{{
+    config(materialized = 'materialized_view')
+}}
+
+SELECT
+  pnc.uuid AS pnc_uuid,
+  pregnancy.uuid AS pregnancy_uuid,
+
+  pnc.patient_id AS patient_id,
+
+  pnc_contact.uuid AS pnc_reported_by,
+  pregnancy_contact.uuid AS pregnancy_reported_by,
+  pnc.reported AS reported,
+  pregnancy.reported AS pregnancy_reported,
+
+  pregnancy.edd as edd,
+  pnc.delivery_date AS delivery_date,
+  pnc.pregnancy_outcome,
+FROM
+  {{ ref('postnatal_care') }} AS pnc
+LEFT JOIN {{ ref('pregnancy') }} ON
+  (pnc.reported > pregnancy.reported AND
+   pnc.reported < (pregnancy.reported + '1 year'::INTERVAL) AND
+   pnc.patient_id = pregnancy.patient_id)
+WHERE
+  (pnc.pregnancy_outcome IN (VALUES ('healthy'), ('still_birth')))
+  AND pnc.patient_id IS NOT NULL AND pnc.patient_id <> ''
+```
+
+## Views vs. Incremental Tables
+
+There are two options for building models using dbt: _views_ and _incremental tables_; see a full description [in the dbt documentation](https://docs.getdbt.com/docs/build/materializations).
+In short, views are normal SQL views, and incremental tables are actual tables, which are periodically updated as new data is created, updated or deleted.
+
+For CHT dbt models, it is generally preferred to start with materialized views. Materialized views are refreshed on every `dbt run`, columns can still be indexed, and they don't require much extra configuration. Non-materialized or non-incremental tables are possible but not recommended because they are unavailable during `dbt run` and anything that depends on them (including dashboards) will also be unavailable. Some base models use incremental tables to ensure good performance for `dbt run`.
+
+To convert a view to an incremental table, change the materialization to `incremental` and add a condition to the `WHERE` clause:
+
+```sql
+{{
+    config(materialized = 'incremental')
+    unique_key='uuid',
+    indexes=[
+      {'columns': ['"uuid"'], 'type': 'hash'},
+      {'columns': ['"saved_timestamp"'], 'type': 'btree'}
+    ]
+}}
+
+
+SELECT
+  uuid,
+  "saved_timestamp",
+
+  ...
+
+FROM
+  {{ ref("data_record") }} data_record
+WHERE
+  form = 'pregnancy'
+
+{% if is_incremental() %}
+  AND "data_record.saved_timestamp" >= {{ max_existing_timestamp('"saved_timestamp"') }}
+{% endif %}
+```
+
+To be performant, the table this model is reading from needs to have the `saved_timestamp` column indexed.
```

Comment:
```suggestion
To be performant, the table from this model is reading form needs to have the `saved_timestamp` column indexed.
```

I am not very sure about this sentence, nor about my suggestion. Can you please review it?